js/src/components/google-account-card/useSwitchGoogleAccount.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { glaData } from '.~/constants';
+import { API_NAMESPACE } from '.~/data/constants';
+import useGoogleAuthorization from '.~/hooks/useGoogleAuthorization';
+import useDispatchCoreNotices from '.~/hooks/useDispatchCoreNotices';
+import useApiFetchCallback from '.~/hooks/useApiFetchCallback';
+
+/**
+ * A hook that returns a handler that initiates Google account disconnect and connect, to support switching to a different Google account.
+ *
+ * The `handleSwitch` handler is meant to be used in button click handler. Upon button click, the handler will:
+ *
+ * 1. Display an info notice that the process is running and request the users to wait.
+ * 2. Call `DELETE /google/connect` API to disconnect the existing connected Google account.
+ * 3. Call `GET /google/connect` API to get the Google OAuth URL.
+ * 4. Redirect the browser to the URL.
+ * 5. If there is an error in the above process, it will display an error notice.
+ *
+ * @return {Array} `[ handleSwitch, loading ]`
+ * 		- `handleSwitch` is meant to be used as button click handler.
+ * 		- `loading` is a state to indicate that the process is running.
+ */
+const useSwitchGoogleAccount = () => {
+	const pageName = glaData.mcSetupComplete ? 'reconnect' : 'setup-mc';
```

Comment:
💅 
The reconnection process will not reach here. Probably we could specify `'setup-mc'` directly.

Reply:
Good call. Done in [`4910f38` (#1072)](https://github.com/woocommerce/google-listings-and-ads/pull/1072/commits/4910f3803de860184f7df2b96458007aa85feda3).